contrib/checkdiff.bash

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MIT
+#
+# Copyright (c) 2021 Rangi
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+FILES=`git diff --name-only $1 HEAD`
+
+edited () {
+	echo "$FILES" | grep -Fxq "$1"
+}
+
+dependency () {
+	if edited $1 && ! edited $2; then
+		echo "Edited '$1' but not '$2'!"
+	fi
+}
+
+# Pull requests that edit the first file without the second may be correct,
+# but are suspicious enough to require review.
+dependency include/linkdefs.h    src/rgbds.5
```

Comment:
Tempted to add `dependency src/fix/main.c srx/fix/rgbfix.1` after nearly missing it in another PR.

Reply:
Should we add dependencies to check for object file revision bumping as well? `src/{asm,link}/object.c` regard the object file serialization, so they should probably be accompanied by a change to `include/linkdefs.h`. (Of course, this fails to check *precisely* the rev ID, but maybe we can add finer tests instead.)

Reply:
What do you think about the completion scripts and manual pages when changing the CLI interfaces? (`src/*/main.c`, I guess.)
This would cover your initial comment, too.

Reply:
Sure. (Go ahead and commit any rules you think would make sense here.)